src/EGraphs/saturation.jl

```diff
@@ -123,21 +124,21 @@ end
 instantiate_enode!(bindings, @nospecialize(g::EGraph), p::PatVar)::Id = v_pair_first(bindings[p.idx])
 function instantiate_enode!(bindings, g::EGraph{ExpressionType}, p::PatExpr)::Id where {ExpressionType}
   add_constant_hashed!(g, p.head, p.head_hash)
-
+  n = copy(p.n)
```

Comment:
It seems to me like not copying would be a bug, even in the case of non-multithreading!